frontend/src/pages/RecipientRecord/pages/Monitoring/index.js

```diff
@@ -56,24 +57,30 @@ export default function Monitoring({
     }
   }, [currentPage, history, recipientId, regionId]);
 
-  useEffect(() => {
+  useDeepCompareEffect(() => {
+    const controller = new AbortController();
     async function fetchMonitoringData(slug) {
-      const data = await lookup[slug].fetcher(recipientId, regionId);
-      lookup[slug].setter(data);
-    }
-
-    if (currentPage && ALLOWED_PAGE_SLUGS.includes(currentPage)) {
       setIsAppLoading(true);
       try {
-        fetchMonitoringData(currentPage);
+        const data = await lookup[slug].fetcher(recipientId, regionId, controller.signal);
+        lookup[slug].setter(data);
       } catch (error) {
         // eslint-disable-next-line no-console
         console.error('Error fetching monitoring data:', error);
-        history.push(`${ROUTES.SOMETHING_WENT_WRONG}/${error.status}`);
+        // todo: handle error (but not abort error)
+        // history.push(`${ROUTES.SOMETHING_WENT_WRONG}/${error.status}`);
```

Comment:
is this a future change?

Reply:
nope I was supposed to fix this before putting PR up for review

Reply:
@AdamAdHocTeam this is ready for another look - removed the abort controller because I seem to have fixed whatever issue was causing the double fetching